libs/core/langchain_core/indexing/base.py

```diff
@@ -1,8 +1,414 @@
 from __future__ import annotations
 
+import abc
 import time
 from abc import ABC, abstractmethod
-from typing import Dict, List, Optional, Sequence, TypedDict
+from typing import (
+    Any,
+    AsyncIterable,
+    AsyncIterator,
+    Dict,
+    Generic,
+    Iterable,
+    Iterator,
+    List,
+    Literal,
+    Optional,
+    Sequence,
+    TypeVar,
+    Union,
+)
+
+from typing_extensions import NotRequired, TypedDict
+
+from langchain_core._api import beta
+from langchain_core.documents.base import BaseMedia
+from langchain_core.runnables import run_in_executor
+from langchain_core.utils import abatch_iterate, batch_iterate
+
+
+class Sort(TypedDict):
```

Comment:
nit: I think in general, [Dataclasses](https://docs.python.org/3/library/dataclasses.html) are recommended over TypedDict. Any reason to prefer a TypedDict here?